### PR TITLE
Made models for payment system

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

**Review by criticus — CHANGES_REQUESTED** [review]

File: payment_system/models.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    duration = models.DurationField('тривалість')
+    grace_period = models.DurationField('пільговий період', null=True)
+
+    class Meta:
+        verbose_name = 'підписка'
+
+
+class Invoice(DataOceanModel):
+    paid_at = models.DateField('дата оплати', null=True)
+    info = models.CharField('інформація', max_length=500, null=True)
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:32]: for the accountant (like 'note')

File: payment_system/models.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        verbose_name = 'проект'
+
+
+class Subscription(DataOceanModel):
+    name = models.CharField('назва', max_length=50, unique=True)
+    description = models.CharField('опис', max_length=500, null=True)
+    price = models.IntegerField('ціна')
+    requests_limit = models.IntegerField('максимальна кількість запитів')
+    duration = models.DurationField('тривалість')
+    grace_period = models.DurationField('пільговий період', null=True)
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:24]: done

File: payment_system/models.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    class Meta:
+        verbose_name = 'проект'
+
+
+class Subscription(DataOceanModel):
+    name = models.CharField('назва', max_length=50, unique=True)
+    description = models.CharField('опис', max_length=500, null=True)
+    price = models.IntegerField('ціна')
+    requests_limit = models.IntegerField('максимальна кількість запитів')
+    duration = models.DurationField('тривалість')
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:23]: done

File: payment_system/models.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    class Meta:
+        verbose_name = 'проект'
+
+
+class Subscription(DataOceanModel):
+    name = models.CharField('назва', max_length=50, unique=True)
+    description = models.CharField('опис', max_length=500, null=True)
+    price = models.IntegerField('ціна')
+    requests_limit = models.IntegerField('максимальна кількість запитів')
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:22]: done for the price. Zhenya wants to limit requests even for unlimited subscriptions - for stopping bugs on the client`s side

File: payment_system/models.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    users = models.ManyToManyField(settings.AUTH_USER_MODEL, verbose_name='користувачі')
+
+    class Meta:
+        verbose_name = 'проект'
+
+
+class Subscription(DataOceanModel):
+    name = models.CharField('назва', max_length=50, unique=True)
+    description = models.CharField('опис', max_length=500, null=True)
+    price = models.IntegerField('ціна')
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:21]: done

File: payment_system/models.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.db import models
+from data_ocean.models import DataOceanModel
+
+
+class Project(DataOceanModel):
+    name = models.CharField('назва', max_length=50, unique=True)
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py:7]: done

File: payment_system/models.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    users = models.ManyToManyField(settings.AUTH_USER_MODEL)
+
+
+class Subscription(DataOceanModel):
+    name = models.CharField(max_length=50, unique=True)
+    description = models.CharField(max_length=500, null=True)
+    price = models.SmallIntegerField(default=0)
+    requests_limit = models.IntegerField('limit for API requests from the project')
+    duration = models.SmallIntegerField()
+    grace_period = models.SmallIntegerField(null=True)
```
Comment (criticus) [inline payment_system/models.py:20]: consider to use default=0

File: payment_system/models.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class Invoice(DataOceanModel):
+    paid_at = models.DateField(null=True)
+    info = models.CharField(max_length=500, null=True)
```
Comment (criticus) [inline payment_system/models.py:25]: what is this field for?

File: payment_system/migrations/0001_initial.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        ),
+        migrations.CreateModel(
+            name='Subscription',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('created_at', models.DateTimeField(auto_now_add=True)),
+                ('updated_at', models.DateTimeField(auto_now=True, null=True)),
+                ('deleted_at', models.DateTimeField(null=True)),
+                ('name', models.CharField(max_length=50, unique=True, verbose_name='назва')),
+                ('description', models.CharField(max_length=500, null=True, verbose_name='опис')),
```
Comment (criticus) [inline payment_system/migrations/0001_initial.py:42]: consider to use latin names

File: payment_system/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply payment_system/models.py]: added a comment to the code

File: payment_system/models.py (hunk 7 above)
Comment (IegorSoboliev) [reply payment_system/models.py]: done. For the duration set to 30

File: payment_system/migrations/0001_initial.py (hunk 9 above)
Comment (IegorSoboliev) [reply payment_system/migrations/0001_initial.py]: replaced the migration

**Review by criticus — APPROVED** [review]
It looks good

**Pull request merged**